src/uwtools/config.py

```diff
@@ -582,31 +580,133 @@ def _format_output(self):
+    config_obj = config_class(user_args.input_base_file, log_name=log.name)
+
+    if user_args.config_file:
+        config_file_type = user_args.config_file_type or cli_helpers.get_file_type(
+            user_args.config_file
+        )
+
+        user_config_obj = globals()[f"{config_file_type}Config"](user_args.config_file)
+
+        if config_file_type != infile_type:
+            config_depth = user_config_obj.dictionary_depth(user_config_obj.data)
+            input_depth = config_obj.dictionary_depth(config_obj.data)
+
+            if input_depth < config_depth:
+                log.critical(f"{user_args.config_file} not compatible with input file")
+                raise ValueError("Set config failure: config object not compatible with input file")
+
+        if user_args.compare:
+            log.info(f"- {user_args.input_base_file}")
+            log.info(f"+ {user_args.config_file}")
+            log.info("-" * 80)
+            config_obj.compare_config(user_config_obj)
+            return
+
+        config_obj.update_values(user_config_obj)
+
+    config_obj.dereference_all()
+
+    if user_args.values_needed:
+        set_var: List[str] = []
+        jinja2_var: List[str] = []
+        empty_var: List[str] = []
+        config_obj.iterate_values(config_obj.data, set_var, jinja2_var, empty_var, parent="")
+        log.info("Keys that are complete:")
+        for var in set_var:
+            log.info(var)
+        log.info("")
+        log.info("Keys that have unfilled jinja2 templates:")
+        for var in jinja2_var:
+            log.info(var)
+        log.info("")
+        log.info("Keys that are set to empty:")
+        for var in empty_var:
+            log.info(var)
+        return
+
+    if user_args.dry_run:
+        # Apply switch to allow user to view the results of config instead of
+        # writing to disk.
+        log.info(config_obj)
+        return
+
+    if user_args.outfile:
+        outfile_type = user_args.output_file_type or cli_helpers.get_file_type(user_args.outfile)
+
+        if outfile_type != infile_type:
+            out_object = globals()[f"{outfile_type}Config"]()
+            out_object.update(config_obj)
+
+            # output_depth = out_object.dictionary_depth(out_object.data)
+            input_depth = config_obj.dictionary_depth(config_obj.data)
+
+            # Check for incompatible conversion objects.
+
+            err_msg = "Set config failure: incompatible file types"
+            if (outfile_type == "INI" and input_depth > 2) or (
+                outfile_type == "F90" and input_depth != 2
+            ):
+                log.critical(err_msg)
+                raise ValueError(err_msg)
+
+            # #PM# HOW IS THIS POSSIBLE?
+        # if input_depth > output_depth:
+        #     log.critical(f"{user_args.outfile} not compatible with {user_args.input_base_file}")
+        #     raise ValueError("Set config failure: output object not compatible with input file")
```

Comment:
Nice call. I think we're checking the more important part above in that the maximum allowable input depths for a given output language are not exceeded.

Reply:
@christinaholtNOAA Yeah, I'm going to remove this code and resolve the conversation. I'll be happy to address anything that comes up, but I don't think it will.